server/src/main/java/io/crate/metadata/cluster/DropPartitionsClusterStateTaskExecutor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.metadata.cluster;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.MetadataDeleteIndexService;
+import org.elasticsearch.index.Index;
+
+import io.crate.execution.ddl.tables.DropPartitionsRequest;
+
+public class DropPartitionsClusterStateTaskExecutor extends DDLClusterStateTaskExecutor<DropPartitionsRequest> {
+
+    private final MetadataDeleteIndexService deleteIndexService;
+
+    public DropPartitionsClusterStateTaskExecutor(MetadataDeleteIndexService deleteIndexService) {
+        this.deleteIndexService = deleteIndexService;
+    }
+
+    @Override
+    protected ClusterState execute(ClusterState currentState, DropPartitionsRequest request) throws Exception {
+        Collection<Index> indices;
+        if (request.partitionValues().isEmpty()) {
```

Comment:
How can the values be empty here? Maybe safer to use a list of `PartitionName`s on the request instead? Did you consider this and do you see issues?

Reply:
> How can the values be empty here?
Answering myself, it's also used to drop all partitions.

Reply:
yep!